pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -572,6 +572,7 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
       // Decode message
       StreamDataDecoderResult decodedRow = _streamDataDecoder.decode(messagesAndOffsets.getStreamMessage(index));
       msgMetadata = messagesAndOffsets.getStreamMessage(index).getMetadata();
+      StreamPartitionMsgOffset messageOffset = messagesAndOffsets.getNextStreamPartitionMsgOffsetAtIndex(index);
```

Comment:
It seems `getNextStreamPartitionMsgOffsetAtIndex` returns the offset of the next message for the given index. Should not we use `getMessageOffsetAtIndex(int index)` for getting the current message's offset that we wanted to use in the error messages?

Reply:
Thanks @satishd for pointing this out! Created a follow-up in #12602 and also added logic for tracking filtered events there as discussed offline.